miner/algo_common.go

```diff
@@ -428,6 +445,22 @@ func (envDiff *environmentDiff) commitSBundle(b *types.SimSBundle, chData chainD
 		return fmt.Errorf("incorrect EGP: got %d, expected %d", gotEGP, simEGP)
 	}
 
+	if enforceProfit {
```

Comment:
Q: why do you have a separate flag that's not present for EGP check? Is it only for testing?

Reply:
@dvush suggested adding enforce profit flag for backwards compatibility with algo greedy builder

Reply:
@Ruteri enforcing profit is important when you use profit to actually sort bundles. Its not needed for greedy builder because it uses EGP for ordering. If you don't enforce, then you can simply do the same thing that was done with EGP manipulation before EGP check was introduced.

Reply:
I'm suggesting to perform both checks still, just outside of this function (by greedy and buckets respectively)